docs/contributing.md

```diff
@@ -35,6 +35,38 @@ Adding examples is quite straight forward, so don't feel shy! Here's how to do i
 
 Aaaand you are done! :tada: 
 
+### New Generators
+Generators sits as the core of Modelina, which frames the core concepts of what you can generate. Therefore it's also no small task to create a new one, so dont get discourage, we are here to help you! 
+
+To make it easier to contribute a new generator, and to avoid focusing too much of the internals of Modelina, we created a template generator to get you started. If you encounter discreprencies with the following guide or templates, make sure to raise it as an issue so it can be fixed!
+
+Here is how:
+1. Copy/paste the [template generator](../src/generators/template/) and [tests](../test/generators/template/) and rename it to your generator.
+2. Search and replace within your new generator and test folder for `Template`, `template` and `TEMPLATE` and replace it with your generator name and match the cases. **Make sure you search and replace it with matching case**.
+3. Replace the filenames `Template...` with your generator name.
+4. Add your [generator to the generator index file](../src/generators/index.ts).
+
+Now it's time to adapt the template into what ever it is you are generating:
+1. Adapt the [constraint logic](../src/generators/template/) based on what is allowed within your output. Read more about the constraint logic here.
+2. Add all of the reserved keywords that the models must never generate in the [Constant file](../src/generators/template/Constants.ts).
+3. Adapt/create the first renderers. The template by default include two renderers, one for rendering enums and one for classes, but you can create what ever renderers makes sense. For example in Rust it's not called class but struct, so therefore its called a `StructRenderer`.
+4. Adapt the file generator and the rendering of the complete models to fit your generator.
+
+Time to adapt the tests, cause without tests, it's just an empty promise. The test that is included in the template is really just placeholders, so make sure you adapt them accordingly to your code.
```

Comment:
A note about black box tests might be helpful here! These tests were the most useful in finding corner cases.

Reply:
You are right! 

Also helped me run down a few bugs 😆

Reply:
I ended up adding the following:

```
6. Lastly, we have (arguably) the most important tests, [the BlackBox tests](./development.md#blackbox-testing). They are to ensure that real-world inputs generate usable models that do not contain syntax errors. You can read more about the BlackBox tests [here](./development.md#blackbox-testing).
```

We can then use `development.md#blackbox-testing` to add more details about black-box testing if needed. I hope that suffice 🤞